src/api/ucc.h

```diff
@@ -0,0 +1,1497 @@
+/**
+ * @file ucc.h
+ * @date 2020
+ * @copyright Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_H_
+#define UCC_H_
+
+#include <api/ucc_def.h>
+#include <api/ucc_version.h>
+#include <api/ucc_status.h>
+#include <stdio.h>
+#include <stdbool.h>
```

Comment:
Does this mean that we stick to C99 and is it safe to use C99 bool with C++?

Reply:
@Sergei-Lebedev I don't understand your question, can you elaborate. Remember, we are using this only at the interface not for data transfer.  Just FYI, https://en.cppreference.com/w/cpp/language/types

Reply:
1. Because stdbool is only available in C99 and we have stdbool.h in main header file it means that every applications that uses ucc will have to be compiled with c99 or c11 support.
2. _Bool a.k.a bool in c99 and c++ bool are in fact different types and my question is it safe to rely on it, for instance when ucc is precompiled and different c++ compiler is used to build user application

Reply:
Are you saying this is going to be an issue or just asking? I don’t know if this is going to be an issue. It is only one place we need this. So, it not a huge blocker. We have three options

1. We can be conservative and change bool -> _Bool
2. Other option is to define our own bool UCC_DT_TYPE_BOOL
3. Leave as it is and deal with it later

Reply:
Sep 30th WG Meeting: Change bool to uint32_t